src/main/java/ru/spbstu/calculator/CalculationStrategy.java

```diff
@@ -10,14 +10,52 @@
  */
 public class CalculationStrategy {
 
-    public double frequencyOfNucleotides;
-    public int sequenceLength;
-    public calculationMethod resultFrequency;
+    private final double frequencyOfNucleotides;
+    private final int sequenceLength;
+    private final calculationMethod resultFrequency;
 
+    public static class Builder {
 
-    public CalculationStrategy(double frequencyOfNucleotides, int sequenceLength, calculationMethod resultFrequency) {
+        private double frequencyOfNucleotides = Constants.frequency;
+        private final int sequenceLength;
+        private calculationMethod calculationType = calculationMethod.LOGLIKELIHOOD;
+
+        public Builder(int sequenceLength) {
+            this.sequenceLength = sequenceLength;
+        }
+
+        public Builder frequencyOfNucleotides(double value) {
+            this.frequencyOfNucleotides = value;
+            return this;
+        }
+
+        public Builder calculationType(calculationMethod value) {
+            this.calculationType = value;
+            return this;
+        }
+
+        public CalculationStrategy build() {
+            return new CalculationStrategy(this.frequencyOfNucleotides, this.sequenceLength, this.calculationType);

```

Comment:
minor:
if the building class requires many parameters, it is better to get them from Builder object. For example, `new CalculationStrategy(this);` and the constructor of CalculationStrategy looks like that
```
private CalculationStrategy(Builder builder) {
         this.frequencyOfNucleotides = builder.frequencyOfNucleotides;
         this.sequenceLength = builder.sequenceLength;
         this.resultFrequency = builder.resultFrequency;
     }
```

Reply:
Ok, may change to this later, if more parameters will be required